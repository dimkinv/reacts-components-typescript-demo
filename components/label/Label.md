A simple label:

```js
<Label text="A Button" />
```
