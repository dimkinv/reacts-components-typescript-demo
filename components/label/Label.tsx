import * as React from 'react';
import LabelProps from './LabelProps';
import LabelState from './LabelState';

export default class Label extends React.Component<LabelProps, LabelState>{
    render(){
        const styles = {
          color: 'red',
          fontSize: '24px'
        };
        return (<span style={styles}>{this.props.text}</span>)
    }
}
