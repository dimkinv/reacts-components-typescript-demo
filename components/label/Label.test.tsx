import * as React from "react";
import { render } from "enzyme";
        
import Label from "./Label";
        
describe("Label", () => {
    it("renders correctly ", () => {
        const instance = render(<Label text="I'm Label" />);
        expect(instance).toMatchSnapshot();
    });
});
