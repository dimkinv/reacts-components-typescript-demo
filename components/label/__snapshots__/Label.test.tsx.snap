// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders correctly  1`] = `
<span
  style="color:red;font-size:24px"
>
  I'm Label
</span>
`;
